lib/avro/resolution_canonical_form.rb

```diff
@@ -12,20 +14,47 @@ def to_resolution_form(schema)
 
     private
 
-    # TODO: include aliases once the avro Ruby gem supports them
-    # Note: permitted values for defaults are not enforced here.
-    # That should be done at the schema level, and is not done currently
-    # in the Avro Ruby gem
+    def normalize_schema(schema)
+      if schema.type_sym == :bytes && schema.logical_type == DECIMAL_LOGICAL_TYPE
+        add_logical_type(schema, type: schema.type)
+      else
+        super
+      end
+    end
+
     def normalize_field(field)
-      default_value = if field.default?
-                        { default: field.default }
-                      else
-                        {}
-                      end
-      super.merge(default_value)
+      extensions = {}
+      extensions[:default] = field.default if field.default?
+      if field.respond_to?(:aliases) && field.aliases && !field.aliases.empty?
+        extensions[:aliases] = field.aliases.sort
+      end
+
+      super.merge(extensions)
     end
 
-    # TODO: Override normalize_named_type once the Avro Ruby gem supports aliases
-    # def normalized_named_type(schema, attributes = {})
+    def add_logical_type(schema, serialized)
+      if schema.respond_to?(:logical_type) && schema.logical_type == DECIMAL_LOGICAL_TYPE
+        extensions = { logicalType: DECIMAL_LOGICAL_TYPE }
+        extensions[:precision] = schema.precision if schema.respond_to?(:precision) && schema.precision
+        extensions[:scale] = schema.scale if schema.respond_to?(:scale) && schema.scale
+        serialized.merge(extensions)
+      else
+        serialized
+      end
+    end
+
+    def normalize_named_type(schema, attributes = {})
```

Comment:
Do we need the `respond_to?` checks in here if `schema` will be a `Avro::Schema::NamedSchema`?

Reply:
The `respond_to?(:default)` check is necessary because only the enum _schema_ has a default. I implemented it this way to be permissive if a `default` attribute is added to other schemas in the future. 

It could instead be changed to check that `schema.type_sym = :enum`.

The check for the `:fullname_aliases` is not necessary if we are only supporting Avro 1.10+.